setup/roles.py

```diff
@@ -29,28 +30,27 @@ class RoleSetup(object):
     def __init__(self):
         pass
 
-    @staticmethod
+    @classmethod
     @check_role_on_disk
-    def add_disk_role(storagerouter_ip, diskname, roles, api, min_size=MIN_PARTITION_SIZE):
+    def add_disk_role(cls, storagerouter_ip, diskname, roles, min_size=MIN_PARTITION_SIZE):
+
         """
         Partition and adds roles to a disk
 
-        :param storagerouter_ip: ip address of a existing storagerouter
+        :param storagerouter_ip: guid of an existing storagerouter
```

Comment:
I was in favor of storagerouter_ip being an ip of a storagerouter instead of a guid ;)